src/app/form/form.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, Validators, FormBuilder,ReactiveFormsModule} from '@angular/forms';
+import { User } from '../Details';
+
+@Component({
+  selector: 'app-form',
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.css']
+})
+export class FormComponent implements OnInit 
+{
+  userForm=new FormGroup({userForm:new FormControl})
+  
+  FirstName: string;
```

Comment:
move all the user properties into a object so that it does not clutter the main space of the component object and also its easier to maintain.